src/main/java/Duke.java

```diff
@@ -1,10 +1,172 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
 public class Duke {
+
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String name = "Johnnythesnake";
+        System.out.println("Hello I'm " + name + "\n" + "What can I do for you? Aside from completing your CS2103 project for you");
+        Scanner scanner = new Scanner(System.in);
+        String filename = "tasks.txt";
+        // Create a File object with the filename
+        File file = new File(filename);
+
+        ArrayList<Tasks> tasksList = new ArrayList<>();
+        if (file.exists()) {
+            tasksList = TaskReader.readTasksFromFile(filename);
+            System.out.println(tasksList);
+        }
+        while (true) {
+            System.out.println("Enter a command: ");
+            String command = scanner.nextLine();
+            if (command.equalsIgnoreCase("bye")) { // bye exits the code
+                Exit exit = new Exit();
+                System.out.println(exit.exitMessage());
+                break;
+            } else if (command.equalsIgnoreCase("list")) { //list shows the task list
+                System.out.println("Here are the tasks in your list: ");
+                if (!tasksList.isEmpty()) {
+                    for (int i = 1; i <= tasksList.size(); i++) {
+                        System.out.println(i + "." + tasksList.get(i - 1));
+                    }
+                }
+            } else if (command.startsWith("unmark")) { // unmark the task in question
+                int taskNumber = Integer.parseInt(command.substring(7)) - 1;
+                if (taskNumber < tasksList.size()) {
+                    Tasks task = tasksList.get(taskNumber);
+                    task.setMarked(false);
+                    tasksList.set(taskNumber, task);
+                    System.out.println("OK, I've marked this task as not done yet:\n" + "  " + tasksList.get(taskNumber));
```

Comment:
Perhaps a line break here and other lines longer than 120 chars would be nice to improve readability?